libsplinter/src/runtime/service/lifecycle_executor/store/diesel/mod.rs

```diff
@@ -175,6 +179,90 @@ impl DieselLifecycleStore<diesel::sqlite::SqliteConnection> {
     }
 }
 
+pub struct DieselConnectionLifecycleStore<'a, C>
+where
+    C: diesel::Connection<TransactionManager = AnsiTransactionManager> + 'static,
+    C::Backend: diesel::backend::UsesAnsiSavepointSyntax,
+{
+    connection: &'a C,
+}
+
+impl<'a, C> DieselConnectionLifecycleStore<'a, C>
+where
+    C: diesel::Connection<TransactionManager = AnsiTransactionManager> + 'static,
+    C::Backend: diesel::backend::UsesAnsiSavepointSyntax,
+{
+    pub fn new(connection: &'a C) -> Self {
+        DieselConnectionLifecycleStore { connection }
+    }
+}
+
+#[cfg(feature = "postgres")]
+impl<'a> LifecycleStore for DieselConnectionLifecycleStore<'a, diesel::pg::PgConnection> {
```

Comment:
This seems like it should be in its own commit as well (it's not mentioned in the commit message)

Overall the commit message doesn't cover the purpose of these changes.